### PR TITLE
Filtering

### LINKED ISSUE #118: Check how logbook handles HTTP streams

### LINKED ISSUE #123: Allow to ignore body for certain messages
We should have the possibility to ignore the body for certain types of messages, e.g.:
- application/pdf
- image/*
- application/octet-stream
- ...

Proposal:
```http
Incoming Request: 2d66e4bc-9a0d-11e5-a84c-1f39510f0d6b
GET http://example.org/test HTTP/1.1
Host: localhost
Content-Type: application/octet-stream

<<binary>>
```

This pretty much also helps with #32 (unauthorized) and #118 (streaming) messages.

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #118
Fixes #123
Fixes zalando-incubator/put-it-to-rest#32

- [x] Remove UnauthorizedRawHttpRequest
- [x] Remove UnauthorizedHttpRequest

### CODE REVIEW

File: logbook-api/src/main/java/org/zalando/logbook/BodyReplacer.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import java.util.function.Predicate;
+
+@FunctionalInterface
+public interface BodyReplacer<T extends BaseHttpMessage> {
+
+    @Nullable
+    String replace(final T message);
+
+    static <T extends BaseHttpMessage> BodyReplacer<T> replaceBody(final Predicate<T> predicate, final String replacement) {
+        return message -> predicate.test(message) ? replacement : null;
```
Comment (lukasniemeier-zalando) [inline logbook-api/src/main/java/org/zalando/logbook/BodyReplacer.java]: move to implementation?

File: logbook-core/src/main/java/org/zalando/logbook/HeaderFilters.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    HeaderFilters() {
+        // package private so we can trick code coverage
+    }
+
+    public static HeaderFilter defaultValue() {
+        return authorization();
+    }
+
+    public static HeaderFilter authorization() {
+        return replaceHeaders("Authorization"::equalsIgnoreCase, "XXX");
```
Comment (lukasniemeier-zalando) [inline logbook-core/src/main/java/org/zalando/logbook/HeaderFilters.java]: What about `JSESSIONID`?

**Comment by lukasniemeier-zalando** [comment]
:+1:

File: logbook-core/src/main/java/org/zalando/logbook/BodyFilters.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        // package private so we can trick code coverage
+    }
+
+    public static BodyFilter defaultValue() {
+        return accessToken();
+    }
+
+    public static BodyFilter accessToken() {
+        final Predicate<String> json = MediaTypeQuery.compile("application/json", "application/*+json");
+        final Pattern pattern = Pattern.compile("(\"(?:access_token|open_id|id_token)\"\\s*\\:\\s*)\".+?\"");
```
Comment (jhorstmann) [inline logbook-core/src/main/java/org/zalando/logbook/BodyFilters.java]: Nitpick: Would not work if the token itself can contain (encoded) quotes :)

**Comment by jhorstmann** [comment]
:+1:

**Comment by lukasniemeier-zalando** [comment]
:+1:

**Pull request merged**